packages/squiggle-lang/src/js/index.ts

```diff
@@ -58,6 +59,23 @@ export function run(
   return resultMap(res, (x) => createTsExport(x, e));
 }
 
+export function parse(
+  squiggleString: string
+): result<null, Extract<errorValue, { tag: "RESyntaxError" }>> {
+  const maybeExpression = parseRescript(squiggleString);
+  if (maybeExpression.tag === "Ok") {
+    return Ok(null); // TODO - return AST
+  } else {
+    if (
+      typeof maybeExpression.value !== "object" ||
+      maybeExpression.value.tag !== "RESyntaxError"
+    ) {
+      throw new Error("Expected syntax error");
```

Comment:
This is a bit hacky but I didn't want to force the consumer to check for all error types. Should we rethink error hierarchy so that `parse` could return only syntax errors?

Reply:
lots of options re polymorphic variants to work with subsets of errors.